### PR TITLE
pos-print: store configuration in mongodb

### LINKED ISSUE #18: pos-print: implement datastore
Datastore should be implemented and replace the in memory implementations with persistence DAO, this will ensure that the data will not be lost if power is turned off.

### PR DESCRIPTION
**Pull request by gadjevb**
Added persistent datastore implementation for cash register devices, thus allowing to save devices indefinitely.

Fixes #18

### CODE REVIEW

File: pos-print/src/main/java/com/clouway/pos/print/adapter/http/HttpBackend.java [hunk 1]
```diff
@@ -31,6 +33,17 @@ public void start() {
...
+    MongoClientOptions opt = MongoClientOptions
+      .builder()
+      .serverSelectionTimeout(5000)
+      .connectTimeout(300)
+      .socketTimeout(1000)
+      .maxWaitTime(500)
+      .connectionsPerHost(3)
+      .build();
+
+    MongoClient client = new MongoClient("localhost:27017", opt);
```
Comment (mgenov) [inline pos-print/src/main/java/com/clouway/pos/print/adapter/http/HttpBackend.java]: hostname should be passed as argument.

File: pos-print/src/test/java/com/clouway/pos/print/persistent/InMemoryDatastoreRule.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.clouway.pos.print.persistent
+
+import com.github.fakemongo.Fongo
+import com.mongodb.client.MongoDatabase
+import org.junit.rules.ExternalResource
+
+/**
+ *@author Borislav Gadjev <borislav.gadjev@clouway.com>
+ */
+class InMemoryDatastoreRule : ExternalResource() {
```
Comment (mgenov) [inline pos-print/src/test/java/com/clouway/pos/print/persistent/InMemoryDatastoreRule.kt]: DatastoreRule as test doesn't care whether the database is persistent or not.

Comment (gadjevb) [inline pos-print/src/test/java/com/clouway/pos/print/persistent/InMemoryDatastoreRule.kt]: This is used for Fongo, how about FongoDatastoreRule?

File: pos-print/src/main/java/com/clouway/pos/print/adapter/http/HttpBackend.java (hunk 1 above)
Comment (gadjevb) [inline pos-print/src/main/java/com/clouway/pos/print/adapter/http/HttpBackend.java]: Like CommandCLI?
```java
class CommandCLI {

  @Argument(prefix = "--", description = "HTTP Listen Address")
  private Integer httpPort = 3030;

  @Argument(prefix = "--", description = "Database hosts", delimiter = ";")
  private String[] dbHost = new String[]{"localhost:27017"};

  @Argument(prefix = "--", description = "Database name")
  private String dbName = "provisioning_test";
```

File: pos-print/src/test/java/com/clouway/pos/print/persistent/InMemoryDatastoreRule.kt (hunk 2 above)
Comment (mgenov) [inline pos-print/src/test/java/com/clouway/pos/print/persistent/InMemoryDatastoreRule.kt]: Why I want to be concrete ? 

If implementation is changed then name have to be changed and all tests will be affected which are using this class.

File: pos-print/src/main/java/com/clouway/pos/print/adapter/http/HttpBackend.java (hunk 1 above)
Comment (mgenov) [inline pos-print/src/main/java/com/clouway/pos/print/adapter/http/HttpBackend.java]: Yes, something like that. 

Kotlin has something similar:
https://github.com/leprosus/kotlin-cli

File: pos-print/src/test/java/com/clouway/pos/print/persistent/InMemoryDatastoreRule.kt (hunk 2 above)
Comment (gadjevb) [inline pos-print/src/test/java/com/clouway/pos/print/persistent/InMemoryDatastoreRule.kt]: Ok, I though I already had DatastoreRule.kt, but it appears I dont't. DatastoreRule it is.

**Comment by gadjevb** [comment]
@mgenov made the required changes!

**Pull request merged**